tools/Benchmarks/Benchmarks.cs

```diff
@@ -0,0 +1,194 @@
+        _enumerableMixed = GetEnumerableResults(ResultType.Mixed);
+        _enumerableTMixed = GetEnumerableTResults(ResultType.Mixed);
+        _readOnlyListAllOk = GetReadOnlyListResults(ResultType.AllOk);
+        _readOnlyListTAllOk = GetReadOnlyListTResults(ResultType.AllOk);
+        _readOnlyListAllFailed = GetReadOnlyListResults(ResultType.AllFailed);
+        _readOnlyListTAllFailed = GetReadOnlyListTResults(ResultType.AllFailed);
+        _readOnlyListMixed = GetReadOnlyListResults(ResultType.Mixed);
+        _readOnlyListTMixed = GetReadOnlyListTResults(ResultType.Mixed);
+    }
+
+    private IEnumerable<Result> GetEnumerableResults(ResultType resultType)
+    {
+        var (min, max) = GetMinMax(resultType);
+        var random = new Random(Seed);
+        var enumerableAllOk = new List<Result>();
+        for (var index = 0; index < Iterations; index++)
+            enumerableAllOk.Add(random.Next(min, max) == 0 ? Result.Ok() : Result.Fail(index.ToString()));
+        return enumerableAllOk.AsEnumerable();
+    }
+
+    private IEnumerable<Result<string>> GetEnumerableTResults(ResultType resultType)
+    {
+        var (min, max) = GetMinMax(resultType);
+        var random = new Random(Seed);
+        var enumerableTAllOk = new List<Result<string>>();
+        for (var index = 0; index < Iterations; index++)
+            enumerableTAllOk.Add(random.Next(min, max) == 0 ? Result.Ok(index.ToString()) : Result.Fail<string>(index.ToString()));
+        return enumerableTAllOk.AsEnumerable();
+    }
+
+    private IReadOnlyList<Result> GetReadOnlyListResults(ResultType resultType)
+    {
+        var (min, max) = GetMinMax(resultType);
+        var random = new Random(Seed);
+        var readOnlyListAllOk = new List<Result>();
+        for (var index = 0; index < Iterations; index++)
+            readOnlyListAllOk.Add(random.Next(min, max) == 0 ? Result.Ok() : Result.Fail(index.ToString()));
+        return readOnlyListAllOk.AsReadOnly();
+    }
+
+    private IReadOnlyList<Result<string>> GetReadOnlyListTResults(ResultType resultType)
+    {
+        var (min, max) = GetMinMax(resultType);
+        var random = new Random(Seed);
+        var readOnlyListTAllOk = new List<Result<string>>();
+        for (var index = 0; index < Iterations; index++)
+            readOnlyListTAllOk.Add(random.Next(min, max) == 0 ? Result.Ok(index.ToString()) : Result.Fail<string>(index.ToString()));
+        return readOnlyListTAllOk.AsReadOnly();
+    }
+
+    private static (int Min, int Max) GetMinMax(ResultType resultType)
+    {
+        int min;
+        int max;
+        switch (resultType)
+        {
+            case ResultType.AllOk:
+                min = 0;
+                max = 0;
+                break;
+            case ResultType.AllFailed:
+                min = 1;
+                max = 1;
+                break;
+            case ResultType.Mixed:
+                min = 0;
+                max = 1;
```

Comment:
`random.Next()`'s max parameter is exclusive, so this should be 2 - your existing benchmarks for mixed results woul've been the same as those for All OK